pennylane/beta/plugins/default_tensor.py

```diff
@@ -191,86 +173,168 @@ def _add_node(self, A, wires, name="UnnamedNode", key="state"):
+                if len(tensor.shape) != len(wires_seq):
+                    raise ValueError(
+                        "Tensor provided has shape={}, which is incompatible "
+                        "with provided sequence of wires {}.".format(tensor.shape, wires_seq)
+                    )
+                tensor = self._expand_dims(tensor, [0, -1])
+                if tensor.shape == (1, 2, 1):
+                    # MPS form
+                    node = self._add_node(tensor, wires=wires_seq, name=name)
+                    nodes.append(node)
+                else:
+                    # break down non-factorized tensors into MPS form
+                    if max(wires_seq) - min(wires_seq) != len(wires_seq) - 1:
+                        raise NotImplementedError(
+                            "Multi-wire state initializations only supported for tensors on consecutive wires."
+                        )
+                    DV = tensor
+                    for idx, wire in enumerate(wires_seq):
+                        if idx < len(wires_seq) - 1:
+                            node = tn.Node(DV)
+                            U, DV, _error = tn.split_node(node, node[:2], node[2:])
+                            node = self._add_node(U, wires=[wire], name=name)
+                        else:
+                            # final wire; no need to split further
+                            node = self._add_node(DV, wires=[wire], name=name)
+                        nodes.append(node)
+            self.mps = tn.matrixproductstates.finite_mps.FiniteMPS(nodes, canonicalize=False)
+            self._free_wire_edges = [node[1] for node in self.mps.nodes]
+
+    def _get_operator_matrix(self, operation, par):
+        """Get the operator matrix for a given operation or observable.
+
+        Args:
+          operation (str): name of the operation/observable
+          par (tuple[float]): parameter values
         Returns:
-            tn.Edge: the newly created edge
+          array: matrix representation.
         """
-        edge = tn.connect(node1[idx1], node2[idx2])
-        return edge
+        A = self._operation_and_observable_map[operation]
+        if not callable(A):
+            return self._array(A, dtype=self.C_DTYPE)
+        return self._asarray(A(*par), dtype=self.C_DTYPE)
 
     def apply(self, operation, wires, par):
-        if operation == "QubitStateVector":
-            state = self._array(par[0], dtype=self.C_DTYPE)
-            if state.ndim == 1 and state.shape[0] == 2 ** self.num_wires:
-                tensor = self._reshape(state, [2] * self.num_wires)
-                self._clear_network_data()
-                self._add_initial_state_nodes(
-                    [tensor], [list(range(self.num_wires))], ["QubitStateVector"]
-                )
-            else:
-                raise ValueError("State vector must be of length 2**wires.")
+        if operation == "QubitStateVector" or operation == "BasisState":
```

Comment:
This was suggested to me by the linter (had a similar one today :smile: )
```suggestion
        if operation in ("QubitStateVector", "BasisState"):
```

Reply:
good call, fixed